ldap/servers/plugins/memberof/memberof.c

```diff
@@ -2894,14 +2909,38 @@ memberof_fixup_task_thread(void *arg)
                           fixup_progress_count, slapi_current_rel_time_t() - fixup_start_time);
     slapi_task_inc_progress(task);
 
+    /* Cleanup task linked list */
+    PR_Lock(fixup_lock);
+    mo_fixup_ll *prev = NULL;
+    for (mo_fixup_ll *curr = fixup_list; curr; curr = curr->next) {
+        mo_fixup_ll *next = curr->next;
+        if (slapi_sdn_compare(curr->sdn, sdn) == 0 &&
+            strcasecmp(curr->filter_str, td->filter_str) == 0)
+        {
+            if (prev == NULL) {
+                /* first node */
+                fixup_list = next;
+                slapi_sdn_free(&curr->sdn);
```

Comment:
I think those 3 lines are common, whether prev==NULL or not. You may move them after the test.

Reply:
I disagree, but could be wrong.  If prev is NULL, then this is the first node, and we need to reset the global head pointer (fixup_list).  We would not want to do that is "prev" has been set.

Reply:
Oh wait, I think this was github issue.  I think you mean the three lines below this (the ones that free curr).  Yes those can go above.  Revising fix...